### ISSUE TITLE
Genotype ploidy summary functions

### ISSUE THREAD

**Issue by timothymillar** [pystatgen/sgkit#416]
It would be useful to be able to summarize ploidy along different dimensions when working with mixed ploidy data.
Currently ploidy can only be inferred from the call_genotype array by the number of allele values > -1 at each call.

Methods could be implemented to calculate the following arrays:
- `'call_genotype_ploidy'` of shape (variants, samples)
- `'variant_ploidy'` of shape (variants, )
- `'sample_ploidy'` of shape (samples, )

If ploidy is variable across a given variant or sample then -1 should be returned indicating mixed-ploidy along that dimension.

There was some discussion of this in #243 but it was agreed that this is better as a separate PR [#373 (comment)](https://github.com/pystatgen/sgkit/pull/373#pullrequestreview-524217735)

Example:
call_genotype array:
```
[[[0, 0, -1, -1], [0, 0,  1,  1], [0, 0, 1, 1]],
 [[0, 0,  0, -1], [0, 1, -2, -2], [0, 1, 1, 1]],
 [[0, 0,  0,  1], [0, 0, -2, -2], [0, 0, 0, 0]],]
```
call_genotype_ploidy array:
```
[[4, 4, 4],
 [4, 2, 4],
 [4, 2, 4]]
```
variant_ploidy array:
```
[ 4,
 -1,
 -1]
```
sample_ploidy array:
```
[4, -1, 4]
```

**Comment by jeromekelleher** [comment]
This seems very sensible to me @timothymillar. I guess one might also reasonably like to know what the maximum ploidy is, so should we add an option to return this, rather than -1 in the mixed case? Or, perhaps we could return (-1 * maximum_ploidy) if it's mixed? Maybe this is trying to be too clever though, and would be confusing/lead to errors.

**Comment by timothymillar** [comment]
It could be nice to return something more informative where the ploidy is variable, however I think I'd prefer to keep it simple for now. One use case for these methods/arrays is for when another statistic needs to check for consistent ploidy across a specific dimension. So keeping the nan/false value simple is probably easier.

The max/min ploidy can be attained with standard methods e.g `call_genotype_ploidy.max(axis=...)`. The proposed methods are really to fill a gap in the generic methods "if all values are equal return value, else return nan/-1".

**Comment by jeromekelleher** [comment]
Thanks @timothymillar, your approach SGTM.

**Comment by tomwhite** [comment]
Can this be closed now that #472 is merged?

**Comment by timothymillar** [comment]
@tomwhite I actually realized a couple of days ago that the methods in #472 still need to be added to the docs and top level API.
I'll put another PR together shortly.
